libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1719,3 +1740,189 @@ Type const& IRGeneratorForStatements::type(Expression const& _expression)
 	solAssert(_expression.annotation().type, "Type of expression not set.");
 	return *_expression.annotation().type;
 }
+
+bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
+{
+	solAssert(!m_returnInfo.has_value(), "");
+
+	m_returnInfo.reset();
+	_tryStatement.externalCall().accept(*this);
```

Comment:
The external call can contain external calls again. If you want to make it extra tight, store the AST if of the call node in the return info and assert that it matches after accepting.

Reply:
@chriseth: The calling AST node (FunctionCall) is already (optionally) stored in ReturnInfo; I've added a solAssert() underneath to ensure it contains the expected one.